File: webapp/psyconnect-web/src/app/models/swipe-card.ts
export interface Therapist {
  profileId: string;
  address?: string;
  languages?: string[];
  specialization?: string[];
  consultationModes?: string[];
  experience?: number;
  rating?: number;
  currency: string;
  ragePrice?: number;
  isAvailable?: boolean;
  availability?: {
    days?: string[];
    timeSlots?: string[];
  };
  currentSession?: string[];
  matchedClients?: string[];
  avatarOverride?: string;
  name: string;
  professionalInfo?: ProfessionalInfo;
}

export interface ProfessionalTitle {
  code: string;
  display: string;
}

export interface Degree {
  type: string;
  field: string;
  institution: string;
  year: number;
}

export interface Certification {
  name: string;
  issuer: string;
  year: number;
}

export interface ProfessionalInfo {
  title: ProfessionalTitle;
  degrees?: Degree[];
  certifications?: Certification[];
  experienceYears?: number;
}
export function mapTherapistResponse(apiData: any): Therapist[] {
  if (!apiData?.data || !Array.isArray(apiData.data)) return [];

  return apiData.data.map((item: any) => ({
    profileId: item.profile_id,
    name: item.name,
    address: item.address,
    languages: item.languages,
    specialization: item.specialization,
    consultationModes: item.consultation_modes,
    experience: item.experience,
    rating: item.rating,
    currency: item.currency,
    ragePrice: item.ragePrice,
    isAvailable: item.is_available,
    avatarOverride: item.avatar_override,
    availability: item.availability
      ? {
          days: item.availability.days,
          timeSlots: item.availability.time_slots,
        }
      : undefined,
    currentSession: [],
    matchedClients: [],
    professionalInfo: item.professional_info
      ? {
          title: item.professional_info.title,
          degrees: item.professional_info.degrees,
          certifications: item.professional_info.certifications,
          experienceYears: item.professional_info.experience_years,
        }
      : undefined,
  }));
}

File: services/mobileflutter/mobileapp/lib/models/swipe_card.dart
class SwipeCard {
  final String profileId;
  final String name;
  final String address;
  final List<String> languages;
  final List<String> specialization;
  final List<String> consultationModes;
  final int experience;
  final double rating;
  final String currency;
  final double price;
  final String avatarUrl;
  final double points;
  final List<String> reasons;
  final String title;

  SwipeCard({
    required this.profileId,
    required this.name,
    required this.address,
    required this.languages,
    required this.specialization,
    required this.consultationModes,
    required this.experience,
    required this.rating,
    required this.currency,
    required this.price,
    required this.avatarUrl,
    required this.points,
    required this.reasons,
    required this.title,
  });

  factory SwipeCard.fromJson(Map<String, dynamic> json) {
    String parsedTitle = "Therapist";
    if (json['professional_info'] != null && json['professional_info']['title'] != null) {
      parsedTitle = json['professional_info']['title']['display'] ?? "Therapist";
    }

    return SwipeCard(
      profileId: json['profile_id'] ?? '',
      name: json['name'] ?? 'Mental Health Specialist',
      address: json['address'] ?? '',
      languages: List<String>.from(json['languages'] ?? []),
      specialization: List<String>.from(json['specialization'] ?? []),
      consultationModes: List<String>.from(json['consultation_modes'] ?? []),
      experience: json['experience'] ?? 0,
      rating: (json['rating'] as num?)?.toDouble() ?? 0.0,
      currency: json['currency'] ?? 'VND',
      price: (json['rage_price'] as num?)?.toDouble() ?? 0.0,
      avatarUrl: json['avatar_override'] ?? '',
      points: (json['points'] as num?)?.toDouble() ?? 0.0,
      reasons: List<String>.from(json['reasons'] ?? []),
      title: parsedTitle,
    );
  }
}
